src/effects/native/flangereffect.cpp

```diff
@@ -89,59 +129,114 @@ void FlangerEffect::processChannel(const ChannelHandle& handle,
 
     // The parameter minimum is zero so the exact center of the knob is 2 beats.
     double lfoPeriodParameter = m_pPeriodParameter->value();
-    double lfoPeriodSamples;
+    double lfoPeriodFrames;
     if (groupFeatures.has_beat_length_sec) {
         // lfoPeriodParameter is a number of beats
         lfoPeriodParameter = std::max(roundToFraction(lfoPeriodParameter, 2.0), 1/4.0);
         if (m_pTripletParameter->toBool()) {
             lfoPeriodParameter /= 3.0;
         }
-        lfoPeriodSamples = lfoPeriodParameter * groupFeatures.beat_length_sec * sampleRate;
+        lfoPeriodFrames = lfoPeriodParameter * groupFeatures.beat_length_sec * sampleRate;
     } else {
         // lfoPeriodParameter is a number of seconds
-        lfoPeriodSamples = std::max(lfoPeriodParameter, 1/4.0) * sampleRate;
+        lfoPeriodFrames = std::max(lfoPeriodParameter, 1/4.0) * sampleRate;
+    }
+
+    // When the period is changed, the position of the sound shouldn't
+    // so time need to be recalculated
+    if (pState->previousPeriodFrames != -1.0) {
+        pState->lfoFrames *= lfoPeriodFrames / pState->previousPeriodFrames;
     }
+    pState->previousPeriodFrames = lfoPeriodFrames;
+
+
     // lfoPeriodSamples is used to calculate the delay for each channel
     // independently in the loop below, so do not multiply lfoPeriodSamples by
     // the number of channels.
 
-    CSAMPLE lfoDepth = m_pDepthParameter->value();
+
+    CSAMPLE_GAIN mix = m_pMixParameter->value();
+    const CSAMPLE_GAIN mix_delta = (mix - pState->prev_mix) /
+            (numSamples / kChannels);
+    const CSAMPLE_GAIN mix_start = pState->prev_mix + mix_delta;
+    pState->prev_mix = mix;
+
+    CSAMPLE_GAIN regen = m_pRegenParameter->value();
+    const CSAMPLE_GAIN regen_delta = (regen - pState->prev_regen) /
+            (numSamples / kChannels);
+    const CSAMPLE_GAIN regen_start = pState->prev_regen + regen_delta;
+    pState->prev_regen = regen;
+
+
+    // With and Manual is limited by amount of amplitude that remains from width
+    // to kMaxDelaMS
+    double width = m_pWidthParameter->value();
+    double manual = m_pManualParameter->value();
+    double maxManual = kCenterDelayMs + (kMaxLfoWidthMs - width) / 2;
+    double minManual = kCenterDelayMs - (kMaxLfoWidthMs - width) / 2;
+    manual = math_clamp(manual, minManual, maxManual);
+
+    const double width_delta = (width - pState->prev_width) /
+            (numSamples / kChannels);
+    const double width_start = pState->prev_width + width_delta;
+    pState->prev_width = width;
```

Comment:
We are using this parameter ramping technique in several effects now. Perhaps you could factor it out to a common function.